### ISSUE TITLE
Some of the ideas to start working on

### ISSUE THREAD

**Issue by Smit-create** [codezonediitj/pydatastructs#343]
Some of the ideas to start working on:
Adding a few basic and necessary functions:
1. `upper_bound`: (similar to that of cpp's upper_bound) that should work on sorted `OneDimensionalArray`.
2. `lower_bound`: (similar to that of cpp's lower_bound) that should work on sorted `OneDimensionalArray`
3. `is_sorted`: To check that a given `OneDimensionalArray` is sorted or not.
4. `binary_search`: To search for a given key in sorted `OneDimensionalArray`.

Adding few Datastructes:
1. Adding a class of multiset (similar to cpp's `multiset`)
2. Adding a class of Fenwick tree 
3. Adding a class of Sparse Table
4. Adding a class of StringHashFunction 
5. LCA for DAG/tree

Some more ideas on strings and Graphs can also be found here: https://github.com/codezonediitj/pydatastructs/wiki/Planned-Features-for-v0.0.1

The list will be updated based on the progress of the above ideas. **We don't assign issues to any individual so whoever is willing to work on any of the above ideas (or on your own idea please let us know what you are working on) should discuss how he/she is going the design the API of the function/classes and how will it be useful**. 
For example, if I am willing to work on adding `binary_search` for sorted ODA, then I would show its working in the comments below as:
```py
binary_search(array, start, end):
    # its algo (not necessary to show here in comments just show how the function/class will look like)
    return True if found else False
```

**Comment by Nora2412** [comment]
I would like to work on is_sorted and binary_search as part of my contribution for GSSOC'21.
For example,
is_sorted(array):
          # algo 
          return True if sorted else false

**Comment by Smit-create** [comment]
Yes, you can try raising a PR for that by adding the said functions under https://github.com/codezonediitj/pydatastructs/blob/master/pydatastructs/linear_data_structures/algorithms.py

**Comment by Smit-create** [comment]
Yes, please raise a PR that.

**Comment by Simar05519** [comment]
I would like to work on binary search.

**Comment by Smit-create** [comment]
> I would like to work on binary search.

You can raise a PR for that since no one has worked for it till now.

**Comment by Simar05519** [comment]
ok

**Comment by Jahnavi-Majji** [comment]
Hello, @Smit-create  I would like to work on KMP and Fenwick tree. Can I start working on it?

**Comment by Smit-create** [comment]
Sure!

**Comment by sHiVaNgI821** [comment]
I would like to work on the sparse table class which would be a part of miscellaneous data structures, as part of my contribution for GSSoC'21

**Comment by sHiVaNgI821** [comment]
I would like to work on lowest_common_ancestor (LCA for DAG/Tree) function which would be a part of the algorithms.py file in the graphs directory, as part of my contribution for GSSoC'21.
The function definition goes like:
```
def lowest_common_ancestor(graph: Graph, vertex1: str, vertex2: str, algorithm: str) -> str: 
             # algo
             returns lca_vertex
```
